api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+                toPersistFormPrivilege = originalFormPrivilege;
+            }
+
+            return bahmniFormPrivilegeDao.saveFormPrivilege(toPersistFormPrivilege);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<FormPrivilege> saveFormPrivileges(List<FormPrivilege> formPrivileges) {
+        log.info("Inside BahmniFormPrivilegesServiceImpl -->saveFormPrivileges " + formPrivileges);
+        List<FormPrivilege> resultList = new ArrayList<FormPrivilege>();
+        List<FormPrivilege> privilegesList = new ArrayList<FormPrivilege>();
+        List<FormPrivilege> oldPrivilegeList = new ArrayList<FormPrivilege>();
+
+//        if (formPrivileges.size() == 1 && formPrivileges.get(0).getPrivilegeName().equalsIgnoreCase("")) {
+//            oldPrivilegeList = getAllPrivilegesForForm(formPrivileges.get(1).getFormId());
+//            if (oldPrivilegeList != null && !(oldPrivilegeList.isEmpty())) {
+//                for (int i = 0; i < oldPrivilegeList.size(); i++) {
+//                    resultList.add(bahmniFormPrivilegeDao.deleteFormPrivilege(oldPrivilegeList.get(i)));
+//                }
+//            }
+//            return resultList;
+//        } else {
+
+            Iterator privilegeItr = formPrivileges.iterator();
+            int formId = 0;
+            String formVersion = "";
+            log.info("Inside BahmniFormPrivilegesServiceImpl -->saveFormPrivileges " + privilegeItr);
+            while (privilegeItr.hasNext()) {
+                LinkedHashMap temp = (LinkedHashMap) privilegeItr.next();
+                List<Map.Entry> entrySetList = new ArrayList<Map.Entry>(temp.entrySet());
+                FormPrivilege tempPrivilege = new FormPrivilege();
+                for (Map.Entry tempEntrySet : entrySetList) {
+                    if (tempEntrySet.getKey().toString().equalsIgnoreCase("formId")) {
+                        tempPrivilege.setFormId((Integer) temp.get("formId"));
+                        formId = (Integer) temp.get("formId");
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("editable")) {
+                        tempPrivilege.setEditable((Boolean) temp.get("editable"));
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("privilegeName")) {
+                         String privilegeName = temp.get("privilegeName").toString();
+                         tempPrivilege.setPrivilegeName(privilegeName);
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("viewable")) {
+                        tempPrivilege.setViewable((Boolean) temp.get("viewable"));
+                    }else if (tempEntrySet.getKey().toString().equalsIgnoreCase("formVersion")) {
+                        tempPrivilege.setFormVersion((String) temp.get("formVersion"));
+                        formVersion = tempPrivilege.getFormVersion();
+                    }
```

Comment:
@buvaneswari-arun can this method split into multiple private methods. It makes the method more readable

Reply:
Removed